src/Components/NotificationsProvider.js

```diff
@@ -91,8 +157,10 @@ export default function NotificationsProvider(props) {
         setNotifications,
         showMore,
         hideBadge,
-        moreNotiRequests,
-        setMoreNotiRequests,
+        GetNotis,
+        setLastVisible,
```

Comment:
I notice that `setNotifications` and `setLastVisbile` are both exposed by this interface, but using them correctly requires a knowledge of the implementation of this context.  There might be better encapsulation here if those methods stayed internal implementation details.  I notice that they are only used to reset the state after the popper is exited.  Perhaps the NotificationsContext could instead expose a function `resetPagination` or something like that, which calls those methods and resets the pagination in the right way.  Then, other devs (aka me) can use NotficationsContext without having to understand how it works under the hood.

ex, in NotificationsProvider:
```js
function resetPagination() {
  setNotifications([]);
  setLastVisible();
}

// and then NotificationsContext.Provider could just return `resetPagination`
```

and in NotificationDropMenu:
```js
const [
    notifications,
    showMore,
    hideBadge,
    GetNotis,
    showNewNotiButton,
    displayLatestNotis,
    resetPagination,
  ] = useContext(NotificationsContext);


<Grow onExited={() => resetPagination()} />
```

Reply:
Great comment - thanks for outlining the advantages to keeping the methods internal!